include/oneapi/dpl/pstl/parallel_backend_serial.h

```diff
@@ -42,6 +42,35 @@ __cancel_execution(oneapi::dpl::__internal::__serial_backend_tag)
 {
 }
 
+template <typename _StorageType>
+struct __thread_enumerable_storage
+{
+    template <typename... Args>
+    __thread_enumerable_storage(Args&&... __args) : __storage(std::forward<Args>(__args)...)
+    {
+    }
+
+    std::uint32_t
+    size() const
```

Comment:
I think better to use `std::size_t` for size-linked stuff.

Reply:
From my point of view, if we describe anything about the size, better to use `std::size_t` type.
The exception - only one case when we save size in the memory and want to reduce the amount of memory.
But of course, I don't insist, it's not one place with another approach.

Reply:
I've done this.  You are right that there is no need to squeeze space out of types like these at this level.  Also, for TBB we rely upon their size type, so its possible in some situations it could have been a narrowing type conversion to uint32_t.